lib/finch.ex

```diff
@@ -351,12 +350,84 @@ defmodule Finch do
 
   See `request/3` for more detailed information.
   """
-  @spec request!(Request.t(), name(), keyword()) ::
+  @spec request!(Request.t(), name(), request_opts()) ::
           Response.t()
   def request!(%Request{} = req, name, opts \\ []) do
     case request(req, name, opts) do
       {:ok, resp} -> resp
       {:error, exception} -> raise exception
     end
   end
+
+  @doc """
+  Sends an HTTP request asynchronously, returning a request reference.
+
+  ## Responses
+
+  Response information is sent to the calling process as it is received
+  in `{ref, response}` tuples. Responses include:
+
+    * `{:status, status}` - HTTP response status
+    * `{:headers, headers}` - HTTP response headers
+    * `{:data, data}` - section of the HTTP response body
+    * `{:error, exception}` - an error occurred during the request
+    * `:done` - request has completed successfully
```

Comment:
Should we support trailing headers just in case?

```suggestion
    * `{:done, trailing_headers}` - request has completed successfully
```

?